component/src/main/java/org/wso2/carbon/identity/authenticator/nuxeo/NuxeoAuthenticator.java

```diff
@@ -0,0 +1,404 @@
+        }
+        BufferedReader reader = null;
+        StringBuilder result = null;
+        Integer statusCode;
+        String line;
+        HttpGet httpget = null;
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            httpget = new HttpGet(url);
+            httpget.setHeader(NuxeoAuthenticatorConstants.AUTHORIZATION,
+                    NuxeoAuthenticatorConstants.AUTHENTICATION_BEARER + accessToken);
+            result = new StringBuilder();
+            statusCode=httpclient.execute(httpget).getStatusLine().getStatusCode();
+            reader = new BufferedReader(new InputStreamReader(httpclient.execute(httpget).getEntity().getContent()));
+            while ((line = reader.readLine()) != null) {
+                result.append(line);
+            }
+            if (statusCode == HttpStatus.SC_OK) {
+                if (log.isDebugEnabled()) {
+                    log.debug("logged in User info: " + result.toString());
+                }
+                return result.toString();
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Couldn't get the successful response while getting the " +
+                            "UserInfo for the logged in User. Status code: " + statusCode + " and Response: " +
+                            result.toString());
+                }
+                throw new AuthenticationFailedException(" Couldn't get the successful response while getting the " +
+                        "UserInfo for the logged in User. Status code: " + statusCode + " and Response: " +
+                        result.toString());
+            }
+        } catch (ClientProtocolException e) {
+            throw new AuthenticationFailedException("Exception occurred from the Malformed url while " +
+                    "getting the user Information. Malformed url: " + httpget.getURI(), e);
+        } catch (IOException e) {
+            throw new AuthenticationFailedException("Exception occurred from the connection was aborted. " +
+                    "(eg. invalid hostname, no server listening or internet connection was lost) " +
+                    "while getting the UserInfo. ", e);
+        }
+    }
+
+    /**
+     * Get the user claims from user info.
+     *
+     * @param userInfo User info.
+     * @return Response Map<ClaimMapping, String>.
+     */
+    protected Map<ClaimMapping, String> mapUserInfo(String userInfo) {
+
+        if (userInfo.isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Couldn't get the User info: " + userInfo + " to do the claim mapping.");
+            }
+            log.error("Couldn't get the User info: " + userInfo + " to do the claim mapping.");
+        }
+        Map<ClaimMapping, String> claims = new HashMap<>();
+        JSONObject userInfoInJsonObj = new JSONObject(userInfo);
+        Iterator parentKey = userInfoInJsonObj.keys();
+        while (parentKey.hasNext()) {
```

Comment:
Seems the claim handling logic is a nit different than what we discussed during the review. The logic should be as follows.

- Traverse through the userinfo JSON
- If it's a simple attribute (like id, entity-type) do the normal claim building 
- If it's a JSONArray build the claim without processing the array content
- If it's a JSONObject and the object key is not `properties`, build the claim without processing the object content

Now, if the JSONObject key is `properties`, we specially handle it.

- Traverse through the `properties` object
- If it's a simple attribute (firstName, lastName) build the claim the usual way
- If it's a JSONArray and,
  - array key is not `groups`, build the claim without processing the array content
  - array key is `groups`, get the array elements, and build a comma separated string with the elements and build the claim. Here, the constructed custom dialect claim should be mapped to the local `role` claim
  - If it's a JSONObject build the claim without processing the object content